### ISSUE TITLE
TGVMax card's number is not sufficient

### ISSUE THREAD

**Issue by TitCaz** [tducret/trainline-python#16]
* trainline version: 0.1.1
* Python version: 3.8.6
* Operating System: Windows

### Description

Trying to search some travels with TGVMax card, it seems that it is no more sufficient to give the card's number.
It reports an error, asking to create an account : {"errors":{"passengers":["Veuillez créer un compte pour réserver vos voyages TGVmax"]}}.
It should give me the available trains.

### What I Did
~~~ python
    TGVMaxPassenger = trainline.Passenger(birthdate="01/01/1998")
    TGVMaxPassenger.add_special_card(trainline.TGVMAX, "MyCardNumber")
    results = trainline.search(
    passengers=[TGVMaxPassenger],
	departure_station='Lyon',
	arrival_station='Paris',
	from_date="10/10/2020 12:00",
	to_date="11/10/2020 12:00")
~~~

And it raised the following error :
~~~
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\trainline\__init__.py", line 602, in search
    ret = t.search(
  File "PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\trainline\__    ret = c._post(url=_SEARCH_URL, post_data=post_data)
init__.py", line 81, in _post
    raise ConnectionError(
requests.exceptions.ConnectionError: Status code 422 for url https://www.trainline.eu/api/v5_1/search
{"errors":{"passengers":["Veuillez créer un compte pour réserver vos voyages TGVmax"}}
~~~

PS : Thank's a lot for this beautiful library

**Comment by antoinevth** [comment]
@benoitdemaegdt explain the reasons here : https://github.com/tducret/trainline-python/issues/11#issuecomment-560578710

For now, there is no solution using this library. If you have any idea/skill/time, you can submit a PR :wink: !

**Comment by tducret** [comment]
I feel that the best we have to do is to migrate to thetrainline.com new APIs.

**Comment by benoitdemaegdt** [comment]
> I feel that the best we have to do is to migrate to thetrainline.com new APIs.

@tducret  unfortunately it seems that a trainline business account is required to use this API :(

**Comment by TitCaz** [comment]
I tried to log in, with my personnal trainline account on the old api (https://www.trainline.fr/api/v5_1/account/signin), and I fall on another Exception : "errors":{"email":["Nous n’avons pas trouvé de compte associé à l’adresse e-mail et au mot de passe indiqués."]}}. It seems that there is a new database of accounts linked to this api, or that I didn't see anything...

~~~ python
        data_login = {"id":"1","email":"MYEMAIL","password":"MYPASSWORD", "facebook_id":None,"facebook_token": None,"google_code": None,"concur_auth_code": None,"concur_new_email": None,"concur_migration_type": None,"source": None,"correlation_key": None,"auth_token": None, "user_id": None}
        post_data_login = json.dumps(data_login)
        ret_login = c._post(url="https://www.trainline.fr/api/v5_1/account/signin", post_data=post_data_login)
~~~

**EDIT:**
Accounts' data come from "https://www.trainline.fr/business", it is free to create a business account.
I succeeded to connect before making a searching request, adding the authentication token sent back in the header's searching request. I post the solving code as soon as I fix an ultimate bug which prevent to have more than approximately 10 trains with this method.

**Comment by antoinevth** [comment]
This issue has been solved by #17. We should close it.
